src/core/operators/file_processing_op.py

```
@@ -0,0 +1,344 @@
+        # Add nodes
+        workflow.add_node("file_processor", _file_processing_wrapper)
+        workflow.add_node("llm_analyzer", _llm_analysis_wrapper)
+        
+        # Define the workflow edges
+        workflow.add_edge(START, "file_processor")
+        workflow.add_edge("file_processor", "llm_analyzer")
+        workflow.add_edge("llm_analyzer", END)
+        
+        return workflow.compile()
+
+    async def arun(self, query: str, context: Optional[Dict[str, Any]] = None, sources: Optional[List[str]] = []) -> OperatorResult:
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)